buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -371,6 +371,7 @@ public CompositeByteBuf removeComponents(int cIndex, int numComponents) {
         return this;
     }
 
+    @Override
     public Iterator<ByteBuf> iterator() {
         ensureAccessible();
         List<ByteBuf> list = new ArrayList<ByteBuf>(components.size());
```

Comment:
Possibly a separate issue...but I wonder if we can avoid this list copy operation and just build our own iterator structure?